### ISSUE TITLE
Feature-Idea: Adding support for coconut (functional python dialect)

### ISSUE THREAD

**Issue by henningsway** [quarto-dev/quarto-cli#1343]
Coconut (https://coconut.readthedocs.io/en/latest/FAQ.html) seems to add very useful functional programming tools to python and can be used as a Jupyter kernel.

It could be promising to use this together with R in a Rmd/qmd-File.

**Comment by cscheid** [comment]
You can _sort of_ use this right now:

    ---
    jupyter: coconut
    ---
    
    # Coconut support
    
    We're missing some styling, but it kind of works:
    
    ```{coconut}
    "hello, world!" |> print
    ```

In general, any jupyter kernel should generally work with `quarto`, as long as that specific .qmd file does not mix different languages.

There are a few things we need to add for proper syntax highlighting of `coconut` cells. But if your goal is to get started, this works right now.

**Comment by henningsway** [comment]
Thank you, this looks promising.

For some reason I couldn't yet get this to render/ execute. I've created a fresh project with a virtual environment and installed the coconut package there, which installs the coconut kernel.

For some reason, when I render the example document a kernel from another project seems to be called, which then raises an error. When I call a *qmd-file with Python-chunks, the document renders nicely.

![image](https://user-images.githubusercontent.com/1403304/179169064-df33af13-cea8-4b5d-b817-d5c9774c31bc.png)


Any thoughts how to fix this? :)

**Comment by henningsway** [comment]
Ah, this is already great to see and it also works on my machine. :)

Should it be already possible to evaluate coconut code interactively, say by selecting the coconut kernel for interactive evaluation via Ctrl+Enter?

**Comment by cscheid** [comment]
No, we do not currently support for general kernels in the IDEs and editor modes.
